src/message.rs

```diff
@@ -0,0 +1,63 @@
+// EasySplash - tool for animated splash screens
+// Copyright (C) 2020  O.S. Systems Software LTDA.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use async_std::{
+    fs, io,
+    os::unix::net::{UnixListener, UnixStream},
+    path::PathBuf,
+    prelude::*,
+};
+use log::trace;
+use std::env;
+
+#[derive(Debug)]
+pub(crate) enum Message {
+    /// Send a interrupt request for the animation.
+    Interrupt,
+}
+
+pub(crate) async fn bind_socket() -> Result<UnixListener, io::Error> {
+    let path = unix_socket_file(true).await?;
+    if path.exists().await {
+        fs::remove_file(&path).await?;
+    }
+
+    trace!("Binding to {:?} unix socket", &path);
+    UnixListener::bind(&path).await
+}
+
+pub(crate) async fn send(msg: Message) -> Result<(), io::Error> {
+    trace!("Sending stop request");
+    UnixStream::connect(unix_socket_file(false).await?).await?.write_all(&[msg.into()]).await
+}
+
+impl From<u8> for Message {
+    fn from(v: u8) -> Self {
+        match v {
+            0x1 => Message::Interrupt,
+            _ => unreachable!("invalid message code"),
+        }
+    }
+}
+
+impl From<Message> for u8 {
+    fn from(msg: Message) -> Self {
+        match msg {
+            Message::Interrupt => 0x1,
+        }
+    }
+}
+
+async fn unix_socket_file(create_missing: bool) -> Result<PathBuf, io::Error> {
+    let path = PathBuf::from(
+        env::var("ES_RUNTIME_DIR").unwrap_or_else(|_| String::from("/run/easysplash")),
```

Comment:
This would be better as a command line option.

Reply:
Done.